client/client.go

```diff
@@ -291,22 +286,10 @@ func (c *Client) handleUnilateral() {
 			}
 
 			switch resp.Type {
-			case imap.StatusOk:
-				if c.Infos != nil {
-					go func() {
-						c.Infos <- resp
-					}()
-				}
-			case imap.StatusNo:
-				if c.Warnings != nil {
-					go func() {
-						c.Warnings <- resp
-					}()
-				}
-			case imap.StatusBad:
-				if c.Errors != nil {
+			case imap.StatusOk, imap.StatusNo, imap.StatusBad:
+				if c.Updates != nil {
 					go func() {
-						c.Errors <- resp
+						c.Updates <- resp
```

Comment:
I don't think it's legitimate to put this in a separate go-routine: if sending to the channel blocks and multiple goroutines exist trying to send different items, isn't the channel item ordering undefined again?
I think the library has to block here, and the user must be responsible for consuming from the channel in parallel with the function call (or somehow be very confident their channel buffer size is large enough to avoid deadlock).